src/components/elementView/detailedView/decommissionedPropertiesView.tsx

```diff
@@ -0,0 +1,74 @@
+import { Classes } from "@blueprintjs/core";
+import "@blueprintjs/core/lib/css/blueprint.css";
+import * as React from "react";
+import { RouteComponentProps, withRouter } from "react-router";
+import {
+    ElementObjectType,
+    DecommissionedFieldsTable,
+} from "../../../utils/utils";
+
+interface DecommissionedPropertiesViewProps {
+    data: ElementObjectType;
+}
+export interface AlertState {
+    isDeleteOpen: boolean;
+    fields: Array<string>;
+}
+
+class DecommissionedPropertiesView extends React.PureComponent<
+    RouteComponentProps & DecommissionedPropertiesViewProps,
+    AlertState
+    > {
+    setFieldNamesFromData = () => {
+        let fields: Array<string> = [];
+        Object.keys(this.props.data).forEach((col: string) => {
```

Comment:
This is probably nitpicky, but is there a reason this is written as a for loop instead of just pushing the two fields that you want to push? It seems like a lot of extra iterating but maybe I don't understand what's going on here

Reply:
Tbh that was just pattern matching, but theoretically you could have only one of the two fields present? I think your way would be nicer but I'll prob just leave it